app/sidekiq/delete_worker.rb

```diff
@@ -42,5 +42,34 @@ def delete(record, destination)
     @harvest_report.update(delete_updated_time: Time.zone.now)
   rescue StandardError => e
     Rails.logger.info "DeleteWorker: Delete Excecution error: #{e}" if defined?(Sidekiq)
+    log_delete_worker_error(e, record, destination)
+  end
+
+  def log_delete_worker_error(exception, record, destination)
```

Comment:
Yeah it feels like the pattern for logging an error could be better

Reply:
It could be good to if the code for logging of error wasn't part of the worker itself as well. So like the worker could just call 

something.log_delete_worker_error(pass some details)

and it would be wrapped up nicely